packages/core/src/sync/index.ts

```diff
@@ -430,7 +430,10 @@ export const createSync = async (args: CreateSyncParameters): Promise<Sync> => {
             const chainId = network.chainId;
 
             await Promise.all([
-              args.syncStore.insertBlocks({ blocks: [event.block], chainId }),
+              args.syncStore.insertBlocks({
```

Comment:
Should we also conditionally call these sync store methods based on `event.filters` ? Or are they each smart enough to no-op if there is no data to insert?

Reply:
They each no-op and skip metrics if the data is empty